File: 1561-maximum-number-of-coins-you-can-get/1561-maximum-number-of-coins-you-can-get.py
class Solution:
    def maxCoins(self, piles: List[int]) -> int:
        n, l, r, coins = len(piles), 1, len(piles)-1, 0
        piles.sort(reverse = True)
        while l < r:
            coins += piles[l]
            l += 2
            r -= 1
        return coins

File: 1561-maximum-number-of-coins-you-can-get/1561-maximum-number-of-coins-you-can-get.cpp
class Solution {
public:
    int maxCoins(vector<int>& piles) {
        int n = piles.size(), l = 1, r = n-1, coins = 0;
        sort(piles.rbegin(), piles.rend());
        while (l < r) {
            coins += piles[l];
            l += 2;
            r -= 1;
        }
        return coins;
    }
};